KDCalendar/CalendarView/CalendarView+Delegate.swift

```diff
@@ -32,11 +32,10 @@ extension CalendarView: UICollectionViewDelegateFlowLayout {
         guard let date = self.dateFromIndexPath(indexPath) else { return }
 
         if let index = selectedIndexPaths.index(of: indexPath) {
-
-            delegate?.calendar(self, didDeselectDate: date)
-
-            selectedIndexPaths.remove(at: index)
-            selectedDates.remove(at: index)
+            // We want to prevent deselecting ship date
+//            delegate?.calendar(self, didDeselectDate: date)
```

Comment:
Maybe we could make this more mergeable with another delegate method:

```swift
if delegate?.calendar(self, canDeselectDate: date) ?? false {
  delegate?.calendar(self, didDeselectDate: date)
  selectedIndexPaths.remove(at: index)
  selectedDates.remove(at: index)
}

Reply:
Another "too big of a refactor" where I think the correct generalization would be a `SelectionMode` enum with an `.alwaysOneSelectedDay` option. The delegate method rings false since everything CAN be deselected, but only by selecting something else.